README.md
Скрипт блокировки рекомендательных сервисов 
===================================

Яндекс.Советник и Помощник на сайтах Интернет-магазинов

для браузеров Chrome, Firefox, Opera, Яндекс.Браузер
работает с любой CMS

Installation
------------

The preferred way to install this extension is through [composer](http://getcomposer.org/download/).

Either run

```
php composer.phar require --prefer-dist skeeks/yii2-ya-stopsovetnik "*"
```

or add

```
"skeeks/yii2-ya-stopsovetnik": "*"
```

Configuration app
----------


```php



```

##Links
* [Web site (rus)](https://cms.skeeks.com)
* [Author](https://skeeks.com)
* [ChangeLog](https://github.com/skeeks-semenov/yii2-ya-stopsovetnik/blob/master/CHANGELOG.md)


___

> [![skeeks!](https://gravatar.com/userimage/74431132/13d04d83218593564422770b616e5622.jpg)](https://skeeks.com)
<i>SkeekS CMS (Yii2) — quickly, easily and effectively!</i>  
[skeeks.com](https://skeeks.com) | [cms.skeeks.com](https://cms.skeeks.com)




CHANGELOG.md
CHANGELOG
==============

1.0.0
-----------------
 * https://github.com/SerjoPepper/kick_sovetnik/blob/master/dist/index.min.js
 * ready

1.0.0-alpha
-----------------
 * ready alpha


src/assets/src/ss.js
/*!
 * @author Semenov Alexander <semenov@skeeks.com>
 * @link http://skeeks.com/
 * @copyright 2010 SkeekS (СкикС)
 * @date 22.03.2017
 *
 * Блокирует Яндекс-Советник на страницах вашего интернет-магазина
 *
 */

(function () {

  function patchOnMessage() {
    window.addEventListener('message', function (e) {
      var data;
      if (typeof e.data === 'string') {
         try {
           data = JSON.parse(e.data);
         }
         catch (e) {
           return;
         }
      } else {
        data = e.data;
      }
      if (data && data.type === 'MBR_ENVIRONMENT') {
        e.stopImmediatePropagation();
        e.stopPropagation();
        e.data = {};
      }
    }, true);
  }

  var observer;

  function init () {
    try {
      observer = new MutationObserver(function (records) {
        check(records);
      });
    } catch (e) {

    }
    if (document.body) {
      checkNodes(document.body.children);
    }
  }

  function startObserve () {
    if (!document.body) {
      setTimeout(startObserve, 200);
      return;
    }
    if (observer) {
      observer.observe(document.body, {childList: true});
    }
  }

  function stopObserve () {
    if (observer) {
      observer.disconnect();
      observer = null;
    }
  }

  // Проверки
  function check (records) {
    records.forEach(function (record) {
      var addedNodes = record.addedNodes;
      if (addedNodes && addedNodes.length) {
        checkNodes(addedNodes);
      }
    });
  }

  function checkNodes (nodes) {
    Array.prototype.slice.call(nodes).forEach(function (node) {
      if (isDiv(node) && (hasSovetnikLink(node))) {
        remove(node);
      }
    });
  }

  function createStyleNode(selector, styles) {
    var styleNode = document.createElement('style');
    var css = '';
    for (var k in styles) {
      if (styles.hasOwnProperty(k)) {
        css += k + ':' + styles[k] + ' !important;\n';
      }
    }
    styleNode.type = 'text/css';
    styleNode.appendChild(document.createTextNode(selector + ', ' + selector + ':hover{' + css + '}'));
    return styleNode;
  }

  function addStyleNode(selector, styles) {
    var styleNode = createStyleNode(selector, styles);
    document.body.appendChild(styleNode);
  }

  function propName(s) {
    return s.replace(/-([a-z])/g, function (g) { return g[1].toUpperCase(); });
  }

  // Скрываем яндекс-советник со страницы, возвращаем прежний margin-top для body
  function remove (node) {
    var rootStyles = {
      background: 'transparent',
      transition: 'none',
      'box-shadow': 'none',
      'border-color': 'transparent'
    };

    setTimeout(function () {
      var onMouseEnter = function () {
        addStyleNode('#' + node.id, {
          'pointer-events': 'none'
        });
        node.removeEventListener('mouseover', onMouseEnter, true);
        node.removeEventListener('mouseenter', onMouseEnter, true);
      };
      node.addEventListener('mouseover', onMouseEnter, true);
      node.addEventListener('mouseenter', onMouseEnter, true);
    }, 3e3);

    addStyleNode('#' + node.id, rootStyles);
    addStyleNode('#' + node.id + ' *', {
      opacity: '0',
      'pointer-events': 'none'
    });

    // следим в течении 3 сек за изменением marginTop у html
    var marginObserver = new MutationObserver(function () {
      var marginTop = document.documentElement.style.marginTop;
      if (marginTop && parseInt(marginTop, 10) !== 0) {
        document.documentElement.style.marginTop = '';
      }
    });
    setTimeout(function () {
      marginObserver.disconnect();
      marginObserver = null;
    }, 5e3);
    marginObserver.observe(document.documentElement, {attributes: true, attributeFilter: ['style']});
    document.documentElement.style.marginTop = '';
  }

  function isDiv (node) {
    return node.tagName === 'DIV';
  }

  function hasSovetnikLink (node) {
    return !!node.querySelector('[href*="sovetnik.market.yandex.ru"]');
  }

  function getStyle (node, prop) {
    return window.getComputedStyle(node).getPropertyValue(prop);
  }

  // function removeAttrs (attr, selector) {
  //   var elements = document.querySelectorAll(selector);
  //   for (var i = 0; i < elements.length; i++) {
  //     elements[i].removeAttribute(attr);
  //   }
  // }

  // function removeElements(selector) {
  //   var elements = document.querySelectorAll(selector);
  //   for (var i = 0; i < elements.length; i++) {
  //     elements[i].parentNode.removeChild(elements[i])
  //   }
  // }

  try {
    /*
    removeElements('meta');
    removeAttrs('itemscope', '[itemscope]');
    removeAttrs(
      'property',
      '[property="gr:mpn"],[property="gr:hasEAN_UCC-13"]'
    );
    removeAttrs(
      'itemprop',
      '[itemprop="name"],[property="gr:name"],[itemprop="brand"],[itemprop="productID"],[itemprop="isbn"],[itemprop="mpn"]' +
      ',[itemprop="hasMPN"],[itemprop="model"],[itemprop="model"],[itemprop=name],[itemprop^="gtin"],[itemprop$="category"]' +
      ',[itemprop="price"],[itemprop="offers"],[itemprop="offer"]'
    );
    removeAttrs(
      'itemtype',
      '[itemtype*="schema.org/Product"],[itemtype*="data-vocabulary.org/Product"],[itemtype*="schema.org/Offer"]'
    );
    */

    init();
    startObserve();
    patchOnMessage();
  } catch (e) {
    if (typeof console !== 'undefined') {
      console.error('error while kick sovetnik', e);
    }
  }

})();

src/assets/src/ss2.js
/**
 * Блокирует Яндекс-Советник на страницах вашего интернет-магазина
 * https://github.com/SerjoPepper/kick_sovetnik
 * Если скрипт по каким то причинам перестал работать или что-то ломает на вашем сайте, пожалуйста, создайте тикет:
 * https://github.com/SerjoPepper/kick_sovetnik/issues
 */

(function () {

  function patchOnMessage() {
    window.addEventListener('message', function (e) {
      var data;
      if (typeof e.data === 'string') {
         try {
           data = JSON.parse(e.data);
         }
         catch (e) {
           return;
         }
      } else {
        data = e.data;
      }
      if (data && data.type === 'MBR_ENVIRONMENT') {
        e.stopImmediatePropagation();
        e.stopPropagation();
        e.data = {};
      }
    }, true);
  }

  var observer;

  function init () {
    try {
      observer = new MutationObserver(function (records) {
        check(records);
      });
    } catch (e) {

    }
    if (document.body) {
      checkNodes(document.body.children);
    }
  }

  function startObserve () {
    if (!document.body) {
      setTimeout(startObserve, 200);
      return;
    }
    if (observer) {
      observer.observe(document.body, {childList: true});
    }
  }

  function stopObserve () {
    if (observer) {
      observer.disconnect();
      observer = null;
    }
  }

  // Проверки
  function check (records) {
    records.forEach(function (record) {
      var addedNodes = record.addedNodes;
      if (addedNodes && addedNodes.length) {
        checkNodes(addedNodes);
      }
    });
  }

  function checkNodes (nodes) {
    Array.prototype.slice.call(nodes).forEach(function (node) {
      if (isDiv(node) && (hasSovetnikImage(node))) {
        remove(node);
      }
    });
  }

  function createStyleNode(selector, styles) {
    var styleNode = document.createElement('style');
    var css = '';
    for (var k in styles) {
      if (styles.hasOwnProperty(k)) {
        css += k + ':' + styles[k] + ' !important;\n';
      }
    }
    styleNode.type = 'text/css';
    styleNode.appendChild(document.createTextNode(selector + ', ' + selector + ':hover{' + css + '}'));
    return styleNode;
  }

  function addStyleNode(selector, styles) {
    var styleNode = createStyleNode(selector, styles);
    document.body.appendChild(styleNode);
  }

  function propName(s) {
    return s.replace(/-([a-z])/g, function (g) { return g[1].toUpperCase(); });
  }

  // Скрываем яндекс-советник со страницы, возвращаем прежний margin-top для body
  function remove (node) {
    var rootStyles = {
      background: 'transparent',
      transition: 'none',
      'box-shadow': 'none',
      'border-color': 'transparent'
    };

    setTimeout(function () {
      var onMouseEnter = function () {
        addStyleNode('#' + node.id, {
          'pointer-events': 'none'
        });
        node.removeEventListener('mouseover', onMouseEnter, true);
        node.removeEventListener('mouseenter', onMouseEnter, true);
      };
      node.addEventListener('mouseover', onMouseEnter, true);
      node.addEventListener('mouseenter', onMouseEnter, true);
    }, 3e3);

    addStyleNode('#' + node.id, rootStyles);
    addStyleNode('#' + node.id + ' *', {
      opacity: '0',
      'pointer-events': 'none'
    });

    // следим в течении 3 сек за изменением marginTop у html
    var marginObserver = new MutationObserver(function () {
      var marginTop = document.documentElement.style.marginTop;
      if (marginTop && parseInt(marginTop, 10) !== 0) {
        document.documentElement.style.marginTop = '';
      }
    });
    setTimeout(function () {
      marginObserver.disconnect();
      marginObserver = null;
    }, 5e3);
    marginObserver.observe(document.documentElement, {attributes: true, attributeFilter: ['style']});
    document.documentElement.style.marginTop = '';
  }

  function isDiv (node) {
    return node.tagName === 'DIV';
  }

  function hasSovetnikLink (node) {
    return !!node.querySelector('[href*="sovetnik.market.yandex.ru"]');
  }

  // найти по картинке
  function hasSovetnikImage (node) {
    if(node.nodeName === '#text'){
      return false;
    }
    return !!node.querySelector('[src*="data:image/png;base64,iVBORw0KGgoAAAA"]');
  }

  function getStyle (node, prop) {
    return window.getComputedStyle(node).getPropertyValue(prop);
  }

  // function removeAttrs (attr, selector) {
  //   var elements = document.querySelectorAll(selector);
  //   for (var i = 0; i < elements.length; i++) {
  //     elements[i].removeAttribute(attr);
  //   }
  // }

  // function removeElements(selector) {
  //   var elements = document.querySelectorAll(selector);
  //   for (var i = 0; i < elements.length; i++) {
  //     elements[i].parentNode.removeChild(elements[i])
  //   }
  // }

  try {
    /*
    removeElements('meta');
    removeAttrs('itemscope', '[itemscope]');
    removeAttrs(
      'property',
      '[property="gr:mpn"],[property="gr:hasEAN_UCC-13"]'
    );
    removeAttrs(
      'itemprop',
      '[itemprop="name"],[property="gr:name"],[itemprop="brand"],[itemprop="productID"],[itemprop="isbn"],[itemprop="mpn"]' +
      ',[itemprop="hasMPN"],[itemprop="model"],[itemprop="model"],[itemprop=name],[itemprop^="gtin"],[itemprop$="category"]' +
      ',[itemprop="price"],[itemprop="offers"],[itemprop="offer"]'
    );
    removeAttrs(
      'itemtype',
      '[itemtype*="schema.org/Product"],[itemtype*="data-vocabulary.org/Product"],[itemtype*="schema.org/Offer"]'
    );
    */

    init();
    startObserve();
    patchOnMessage();
  } catch (e) {
    if (typeof console !== 'undefined') {
      console.error('error while kick sovetnik', e);
    }
  }

})();